crud/common/utils.lua

```diff
@@ -199,11 +199,64 @@ function utils.tarantool_supports_uuids()
     return enabled_tarantool_features.uuids
 end
 
-function utils.convert_operations(user_operations, space_format)
-    if utils.tarantool_supports_fieldpaths() then
-        return user_operations
+local function id_not_in_table(operations, id)
+    for _, operation in ipairs(operations) do
+        if operation[2] == id then
+            return false
+        end
+    end
+
+    return true
+end
+
+local function add_nullable_fields_rec(operations, space_format, tuple, id)
+    if id < 2 or tuple[id - 1] ~= box.NULL then
+        return operations
+    end
+
+    if space_format[id - 1].is_nullable and id_not_in_table(operations, id - 1) then
+        table.insert(operations, {'=', id - 1, box.NULL})
+        return add_nullable_fields_rec(operations, space_format, tuple, id - 1)
+    end
+
+    return operations
+end
+
+-- Tarantool < 2.3 has no fields `box.error.NO_SUCH_FIELD_NO` and `box.error.NO_SUCH_FIELD_NAME`.
+if _TARANTOOL >= "2.3" then
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD_NO or err_code == box.error.NO_SUCH_FIELD_NAME
+    end
+else
+    function utils.is_field_not_found(err_code)
+        return err_code == box.error.NO_SUCH_FIELD
+    end
+end
+
+function utils.add_intermediate_nullable_fields(operations, space_format, tuple)
+    if tuple == nil then
+        return operations
+    end
+
+    local formatted_operations, err = utils.convert_operations(operations, space_format)
```

Comment:
If `not utils.tarantool_supports_fieldpaths()`, we have already format the operations.